source/docs/network_filter_fuzzing.md

```diff
@@ -0,0 +1,119 @@
+There are two ways to link it into the fuzzer. 
+* [Recommended] In the file [extensions_build_config.bzl](https://github.com/envoyproxy/envoy/blob/master/source/extensions/extensions_build_config.bzl), the name of the filter should have a prefix `envoy.filters.network`. If it has such a prefix, the filter will be automatically linked into Generic ReadFilter Fuzzer.
+* [Not recommended]If for some reasons the filter's name doesn't have such a prefix, the config of the filter must be added into the `deps` field of `network_readfilter_fuzz_test` module in the file [BUILD](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/BUILD).
+### Step2. Add the filter name into supported_filter_names
+In [uber_per_readfilter.cc](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc), add the filter name into the vector `supported_filter_names` in method `UberFilterFuzzer::filterNames()`.
+```
+const std::vector<absl::string_view> supported_filter_names = {
+...
+NetworkFilterNames::get().ExtAuthorization, NetworkFilterNames::get().TheNewFilterCreatedByYou,
+...
+};
+```
+
+# Add a new WriteFilter into Generic Writefilter Fuzzer
+## Step 1. Make sure the filter can be linked into the fuzzer
+For WriteFilter, the config of the filter must be added into the `deps` field of `network_writefilter_fuzz_test` module in the file [BUILD](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/BUILD).
+```
+envoy_cc_fuzz_test(
+    name = "network_writefilter_fuzz_test",
+    srcs = ["network_writefilter_fuzz_test.cc"],
+    corpus = "network_writefilter_corpus",
+    # All Envoy network filters must be linked to the test in order for the fuzzer to pick
+    # these up via the NamedNetworkFilterConfigFactory.
+    deps = [
+        ":uber_writefilter_lib",
+        "//source/common/config:utility_lib",
+        "//source/extensions/filters/network/kafka:kafka_broker_config_lib",
+        "//source/extensions/filters/network/mongo_proxy:config",
+        "//source/extensions/filters/network/mysql_proxy:config",
+        "//source/extensions/filters/network/zookeeper_proxy:config",
+        "//source/extensions/filters/network/the_new_filter_created_by_you:config", // <---Add the filter config module here
+        "//test/config:utility_lib",
+    ],
+)
+```
+## Step 2. Add the filter name into supported_filter_names
+In [uber_per_writefilter.cc](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/uber_per_writefilter.cc), add the filter name into the vector `supported_filter_names` in method `UberWriteFilterFuzzer::filterNames()`.
+```
+const std::vector<absl::string_view> supported_filter_names = {
+      ...
+      NetworkFilterNames::get().ExtAuthorization, NetworkFilterNames::get().TheNewFilterCreatedByYou,
+      ...
+    };
+```
+
+# Add test cases into corpus
```

Comment:
Mention that this is optional.

I want every developer to do this.  I fear that if we indicate that it is required, some will not bother with fuzzing.

Pointing out that this is an optional step allows developers to start with a simple fuzzer.  We can talk them in to making a better corpus as a next step.

Reply:
> Mention that this is optional.
> 
> I want every developer to do this. I fear that if we indicate that it is required, some will not bother with fuzzing.
> 
> Pointing out that this is an optional step allows developers to start with a simple fuzzer. We can talk them in to making a better corpus as a next step.

I agree. Thanks!

Reply:
> Mention that this is optional.
> 
> I want every developer to do this. I fear that if we indicate that it is required, some will not bother with fuzzing.
> 
> Pointing out that this is an optional step allows developers to start with a simple fuzzer. We can talk them in to making a better corpus as a next step.

If there is no corpus file for this filter, in the CI report, maybe the coverage of this filter is almost 0(when fuzzer didn't find a valid config to pass the validation), because it only runs 1 minutes for at least 18 filters. Does this matter? 
On OSS-fuzz I believe sooner or later the valid config will be found.